protocols/delegate/contracts/Delegate.sol

```diff
@@ -180,6 +166,84 @@ contract Delegate is IDelegate, Ownable {
     );
   }
 
+  /**
+    * @notice sets a rule on the delegate and an intent on the indexer
+    * @dev only callable by owner
+    * @dev delegate needs to be given allowance from msg.sender for the _amountToStake
+    * @dev swap needs to be given permission to move funds from the delegate
+    * @param _senderToken the token the delgeate will send
+    * @param _senderToken the token the delegate will receive 
+    * @param _rule the rule to set on a delegate
+    * @param _amountToStake the amount to stake for an intent
+    * @param _indexer the indexer the intent will be set on
+    */
+  function setRuleAndIntent(
+    address _senderToken,
+    address _signerToken,
+    Types.Rule calldata _rule,
+    uint256 _amountToStake,
+    IIndexer _indexer
+  ) external onlyOwner {
+    
+    _setRuleInternal(
+      _senderToken,
+      _signerToken,
+      _rule.maxSenderAmount, 
+      _rule.priceCoef,
+      _rule.priceExp
+    );
+
+    require(
+      IERC20(_indexer.stakeToken())
+      .allowance(msg.sender, address(this)) >= _amountToStake, "ALLOWANCE_FUNDS_ERROR"
```

Comment:
Same comment as on previous PR but I think we should remove this require statements. We have other `transferFrom` occurrences in the code that don't check this beforehand.

Reply:
There is no explicit reason for the failure when it happens. It just returns
```
Error: Returned error: VM Exception while processing transaction: revert
```
which I didn't feel was descriptive enough from a user standpoint.

I'm on the side of letting the caller know exactly what the problem is and not let them guess.

Reply:
I'll gladly go back into those other areas of code and update them too.

Reply:
Don and I discussed this. This is a check that UIs can do - it's free to check approvals, balances etc off-chain.